crates/core/src/db/relational_db.rs

```diff
@@ -321,15 +322,23 @@ impl RelationalDB {
         &self,
         tx: &mut MutTxId,
         table_id: u32,
-        col_id: u32,
-    ) -> Result<Option<ColumnIndexAttribute>, DBError> {
+        cols: &NonEmpty<u32>,
+    ) -> Result<ColumnIndexAttribute, DBError> {
         let table = self.inner.schema_for_table_mut_tx(tx, TableId(table_id))?;
-        let Some(column) = table.columns.get(col_id as usize) else {
-            return Ok(None);
+        let columns = table.project_not_empty(cols)?;
```

Comment:
Yeah this whole dance is super wonky, but I realize why we have to do this for now. Def need to clean this up.